src/volcanic_island_simulator.py

```diff
@@ -23,6 +24,23 @@
     "spacing": 100.0,  # node spacing, m
 }
 
+_DEFAULT_FLUVIAL_PARAMS = {
+    "grid": grid, # a grid
```

Comment:
Probably should leave out grid from the defaults, since it's undefined at this point in the code; send it as its own parameter (as you've done below).